test/utils/test_gp_sampling.py

```diff
@@ -403,16 +413,19 @@ def test_get_deterministic_model(self):
                     expected_Y = torch.stack(
                         [basis(X) @ w for w, basis in zip(weights, bases)], dim=-1
                     )
-                    self.assertTrue(torch.equal(Y, expected_Y))
+                    self.assertTrue(torch.allclose(Y, expected_Y))
                     self.assertEqual(Y.shape, torch.Size([*batch_shape, 1, m]))
 
     def test_get_deterministic_model_multi_samples(self):
         tkwargs = {"device": self.device}
         n_samples = 5
         for dtype, m in product((torch.float, torch.double), (1, 2)):
             tkwargs["dtype"] = dtype
-            for batch_shape_w, batch_shape_x in product(
-                [torch.Size([]), torch.Size([3])], repeat=2
+            for batch_shape_w, batch_shape_x in (
+                (torch.Size([]), torch.Size([])),
+                (torch.Size([]), torch.Size([3])),
+                (torch.Size([3]), torch.Size([3])),
+                (torch.Size([3]), torch.Size([5, 3])),
```

Comment:
Using `with self.subTest(batch_shape_w=batch_shape_w, batch_shape_x=batch_shape_x):` as the line below would make it easier to figure out where failures are coming from

Reply:
Cool! I didn't know this existed